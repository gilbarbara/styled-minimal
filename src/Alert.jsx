import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

const styles = (props: Object): string => {
  const { size } = props;
  const { borderRadius, maxWidth, padding } = themeGet(props, 'alert');

  return css`
    ${baseStyles.variant};
    border-radius: ${px(borderRadius)};
    font-size: ${baseStyles.fontSize};
    line-height: ${baseStyles.lineHeight};
    max-width: ${px(maxWidth)};
    padding: ${px(padding[size][0])} ${px(padding[size][1])};
    width: 100%;

    a {
      ${baseStyles.color};
    }
  `;
};

const Alert = styled(Box).attrs({
  role: 'alert',
})`
  ${styles};
`;

Alert.displayName = 'Alert';

Alert.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  /** button size */
  size: sizesAllPropTypes,
  /** button variant */
  variant: variantPropTypes,
  ...Box.propTypes,
};

Alert.defaultProps = {
  outline: false,
  size: 'md',
  variant: 'primary',
};

export default Alert;
