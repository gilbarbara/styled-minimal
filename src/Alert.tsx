import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import {
  baseStyles,
  colorStyles,
  lineHeightStyles,
  sx,
  system,
  variantStyles,
} from './utils/system';

import { AlertProps } from './types';

export const StyledAlert = styled('div')<AlertProps>(
  props => {
    const { borderRadius, maxWidth, padding } = getTheme(props, 'alert');

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${px(borderRadius)};
      font-size: inherit;
      line-height: ${lineHeightStyles};
      max-width: ${px(maxWidth)};
      padding: ${padding};
      width: 100%;

      a {
        ${colorStyles};
      }
    `;
  },
  sx,
  system,
);

export const Alert = React.forwardRef<HTMLDivElement, AlertProps>((props, ref) => (
  <StyledAlert ref={ref} {...props} />
));

Alert.defaultProps = {
  dark: false,
  invert: false,
  role: 'alert',
  variant: 'primary',
};
