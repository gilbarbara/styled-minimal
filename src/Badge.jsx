import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

export const Badge = styled(Box)`
  ${(props: Object): string => {
    const {
      borderRadius,
      fontSize,
      fontWeight,
      padding,
    } = themeGet(props, 'badge');
    const fontSizeProp = themeGet(props, 'componentSizes', { key: 'size' });

    return css`
      ${baseStyles.variant};
      border-radius: ${px(borderRadius)};
      display: inline-flex;
      font-size: ${fontSize};
      ${fontSizeProp ? `font-size: ${fontSizeProp}` : ''};
      font-weight: ${fontWeight};
      line-height: 1;
      padding: ${px(padding[0])} ${px(padding[1])};
      vertical-align: baseline;
    `;
  }};
`;

Badge.displayName = 'Badge';

Badge.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  size: sizesAllPropTypes,
  variant: variantPropTypes,
  ...Box.propTypes,
};

Badge.defaultProps = {
  as: 'span',
};

export default Badge;
