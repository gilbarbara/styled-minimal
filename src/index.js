export { default as Alert } from './Alert';
export { default as Badge } from './Badge';
export { default as Box } from './Box';
export { default as Button } from './Button';
export { default as Code } from './Code';
export { default as Container } from './Container';
export { default as Embed } from './Embed';
export { default as Flex } from './Flex';
export { default as Fieldset } from './Fieldset';
export { default as Form } from './Form';
export { default as FormGroup } from './FormGroup';
export { default as Group } from './Group';
export { default as Heading } from './Heading';
export { default as Image } from './Image';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Legend } from './Legend';
export { default as Link } from './Link';
export { default as List } from './List';
export { default as Paragraph } from './Paragraph';
export { default as Screen } from './Screen';
export { default as Select } from './Select';
export { default as Switch } from './Switch';
export { default as Table } from './Table';
export { default as Text } from './Text';
export { default as Textarea } from './Textarea';
