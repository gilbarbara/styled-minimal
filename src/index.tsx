export { Alert } from './Alert';
export { Badge } from './Badge';
export { Box } from './Box';
export { Button } from './Button';
export { ButtonGroup } from './ButtonGroup';
export { Code } from './Code';
export { CodeBlock } from './CodeBlock';
export { Container } from './Container';
export { Embed } from './Embed';
export { Flex } from './Flex';
export { Fieldset } from './Fieldset';
export { Form } from './Form';
export { FormGroup } from './FormGroup';
export { Grid } from './Grid';
export { Group } from './Group';
export { Heading } from './Heading';
export { Image } from './Image';
export { Checkbox, Input, Radio } from './Input';
export { Label } from './Label';
export { Legend } from './Legend';
export { Link } from './Link';
export { DL, OL, UL } from './List';
export { Loader } from './Loader';
export { Paragraph } from './Paragraph';
export { RadioGroup } from './RadioGroup';
export { Select } from './Select';
export { Switch } from './Switch';
export { Table } from './Table';
export { Text } from './Text';
export { Textarea } from './Textarea';

export { default as system } from '@styled-system/css';
export { mergeTheme as getTheme, px, responsive } from './utils/helpers';
