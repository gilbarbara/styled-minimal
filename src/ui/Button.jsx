import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { getProp } from '../utils/helpers';
import { colors } from '../utils/styles';

const Wrapper = styled.button.attrs({
  type: 'button',
})`
  align-items: center;
  border-radius: ${getProp('buttonRadius', 'size', 'md')};
  background-color: ${getProp('palette', 'variant', 'primary')};
  border: 0;
  color: ${({ variant }) => (variant === 'default' ? colors.black : colors.white)};
  display: flex;
  font-size: ${getProp('fontSizes', 'size', 'md')};
  font-weight: 700;
  padding: ${getProp('buttonPadding', 'size', 'md')};
  text-transform: uppercase;
  ${({ loading }) => (loading ? getProp('buttonLoader') : '')}
  
  [class^=i-] {
    margin-right: 1rem;
  }
`;

const Button = ({ children, ...props }) => (
  <Wrapper {...props}>{children}</Wrapper>
);

Button.propTypes = {
  children: PropTypes.node.isRequired,
  loading: PropTypes.bool,
  size: PropTypes.string,
  variant: PropTypes.string,
};

Button.defaultProps = {
  size: 'md',
  variant: 'primary',
};

export default Button;
