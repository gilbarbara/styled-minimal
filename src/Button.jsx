import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { withComponent } from './utils/helpers';
import { buttonType, sizeTypeFull, variantType } from './utils/propTypes';
import { ButtonStyles } from './utils/system';

import { StyledBadge } from './Badge';

export const StyledButton = styled.button`
  ${ButtonStyles.base};
  
  &:disabled {
    pointer-events: none;
  }
  
  &:hover {
    transition: transform 0.2s;
    transform: scale(1.05);
  }
  
  &:focus {  
      ${ButtonStyles.outlineColor};
  }
  
  ${StyledBadge} {
    margin-left: 5px;
  }
`;

StyledButton.propTypes = {
  /** boolean indicating whether the button should render with an animation */
  animate: PropTypes.bool,
  as: PropTypes.string,
  /** match its parent width */
  block: PropTypes.bool,
  children: PropTypes.node.isRequired,
  /** boolean indicating whether the button should render as disabled */
  disabled: PropTypes.bool,
  /** callback on the click event */
  onClick: PropTypes.func,
  outline: PropTypes.bool,
  size: sizeTypeFull,
  type: buttonType,
  variant: variantType,
};

StyledButton.defaultProps = {
  animate: false,
  block: false,
  disabled: false,
  outline: false,
  size: 'md',
  type: 'button',
  variant: 'primary',
};

export default withComponent(StyledButton, 'Button');
