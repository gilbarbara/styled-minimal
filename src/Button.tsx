import * as React from 'react';
import styled, { css } from 'styled-components';

import { getColor, getTheme, px } from './utils/helpers';
import { baseStyles, sx, system, variantStyles } from './utils/system';

import { StyledBadge } from './Badge';

import { ButtonProps } from './types';

export const StyledButton = styled('button')<ButtonProps>(
  props => {
    const { block, busy, invert, size = 'md' } = props;
    const { borderRadius, disabledOpacity, fontSize, lineHeight, loader, padding } = getTheme(
      props,
      'button',
    );

    return css`
      ${baseStyles};
      ${variantStyles};
      ${busy ? loader(invert ? '#ccc' : '#fff') : ''};
      align-items: center;
      border-radius: ${px(borderRadius[size])};
      box-shadow: none;
      cursor: pointer;
      display: inline-flex;
      font-family: inherit;
      font-size: ${px(fontSize[size])};
      line-height: ${lineHeight};
      padding: ${px(padding[size][0])} ${px(padding[size][1])};
      text-decoration: none;
      width: ${block ? '100%' : 'auto'};

      &:disabled {
        opacity: ${disabledOpacity};
        pointer-events: none;
      }

      &:focus {
        outline-color: ${getColor};
      }

      ${StyledBadge} {
        margin-left: 5px;
      }
    `;
  },
  sx,
  system,
);

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => (
  <StyledButton ref={ref} {...props} />
));

Button.defaultProps = {
  block: false,
  busy: false,
  dark: false,
  disabled: false,
  invert: false,
  size: 'md',
  type: 'button',
  variant: 'primary',
};
