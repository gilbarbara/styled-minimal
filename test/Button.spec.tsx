import * as React from 'react';
import { render } from '@testing-library/react';

import { Button } from '../src/Button';

describe('Button', () => {
  it('should render properly', () => {
    const { container } = render(<Button>CLICK ME</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render properly with `busy`', () => {
    const { container } = render(<Button busy>LOADING</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render properly with `dark`', () => {
    const { container } = render(<Button dark>CLICK ME</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render properly with `invert`', () => {
    const { container } = render(<Button invert>CLICK ME</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render properly with `variant`', () => {
    const { container } = render(<Button variant="green">CLICK ME</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
