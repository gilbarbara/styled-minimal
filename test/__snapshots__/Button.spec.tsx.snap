// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #2E2E2E;
  border: 1px solid #2E2E2E;
  color: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px;
  box-shadow: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
}

.c0:disabled {
  opacity: 0.7;
  pointer-events: none;
}

.c0:focus {
  outline-color: #2E2E2E;
}

.c0 .c1 {
  margin-left: 5px;
}

<button
  class="c0"
  type="button"
>
  CLICK ME
</button>
`;

exports[`Button should render properly with \`busy\` 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #2E2E2E;
  border: 1px solid #2E2E2E;
  color: #fff;
  -webkit-animation: kQznpt 0.5s linear infinite;
  animation: kQznpt 0.5s linear infinite;
  background-image: repeating-linear-gradient( 45deg, rgba(255,255,255,0.4), rgba(255,255,255,0.4) 20px, transparent 20px, transparent 40px );
  background-size: 56px 56px;
  pointer-events: none;
  -webkit-transition: none !important;
  transition: none !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px;
  box-shadow: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
}

.c0:disabled {
  opacity: 0.7;
  pointer-events: none;
}

.c0:focus {
  outline-color: #2E2E2E;
}

.c0 .c1 {
  margin-left: 5px;
}

<button
  class="c0"
  type="button"
>
  LOADING
</button>
`;

exports[`Button should render properly with \`dark\` 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #3C3F41;
  border: 1px solid #3C3F41;
  color: #969696;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px;
  box-shadow: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
}

.c0:disabled {
  opacity: 0.7;
  pointer-events: none;
}

.c0:focus {
  outline-color: #2E2E2E;
}

.c0 .c1 {
  margin-left: 5px;
}

<button
  class="c0"
  type="button"
>
  CLICK ME
</button>
`;

exports[`Button should render properly with \`invert\` 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: transparent;
  border: 1px solid #2E2E2E;
  color: #2E2E2E;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px;
  box-shadow: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
}

.c0:disabled {
  opacity: 0.7;
  pointer-events: none;
}

.c0:focus {
  outline-color: #2E2E2E;
}

.c0 .c1 {
  margin-left: 5px;
}

<button
  class="c0"
  type="button"
>
  CLICK ME
</button>
`;

exports[`Button should render properly with \`variant\` 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #4CAF50;
  border: 1px solid #4CAF50;
  color: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px;
  box-shadow: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
}

.c0:disabled {
  opacity: 0.7;
  pointer-events: none;
}

.c0:focus {
  outline-color: #4CAF50;
}

.c0 .c1 {
  margin-left: 5px;
}

<button
  class="c0"
  type="button"
>
  CLICK ME
</button>
`;
